helix-core/src/indent.rs

```diff
@@ -186,106 +188,291 @@ pub fn indent_level_for_line(line: RopeSlice, tab_width: usize) -> usize {
     len / tab_width
 }
 
-/// Find the highest syntax node at position.
-/// This is to identify the column where this node (e.g., an HTML closing tag) ends.
-fn get_highest_syntax_node_at_bytepos(syntax: &Syntax, pos: usize) -> Option<Node> {
-    let tree = syntax.tree();
+struct AddedIndent {
+    indent: bool,
+    outdent: bool,
+}
+impl AddedIndent {
+    fn new() -> Self {
+        AddedIndent {
+            indent: false,
+            outdent: false,
+        }
+    }
+    fn combine_with(&mut self, other: &AddedIndent) {
+        self.indent |= other.indent;
+        self.outdent |= other.outdent;
+    }
+}
 
-    // named_descendant
-    let mut node = match tree.root_node().descendant_for_byte_range(pos, pos) {
-        Some(node) => node,
-        None => return None,
-    };
+struct IndentResult {
+    indent: i32,
+}
```

Comment:
Can you please add a comment here saying that indentation is in terms of levels and not spaces? Also, I wonder how well this will work for python indention for this:
```python
def hello(a,
          b,
          c):
```

given that the indentation depends on the previous line, if there are no items there.

Reply:
Regarding python, I have this on my radar, I'm planning to add something like an `align` specification. I still have to think about how to make it as intuitive and general as possible. It's kinda difficult to define what to make the "anchor" for the alignment. For simple cases (like aligning the parameters of a function) it should not be to complicated though, since treesitter does all the heavy lifting.